src/main/java/com/rtb/projectmanagementtool/private-comments/PrivateCommentController.java

```diff
@@ -0,0 +1,102 @@
+package com.rtb.projectmanagementtool.privatecomment;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.*;
+import com.rtb.projectmanagementtool.user.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Class controlling the CommentData object. */
+public final class PrivateCommentController {
+
+  private DatastoreService datastore;
+  private static final Filter NO_QUERY_FILTER = null;
+  private static final int NO_QUERY_LIMIT = Integer.MAX_VALUE;
+  private static final SortPredicate NO_QUERY_SORT = null;
+
+  public PrivateCommentController(DatastoreService datastore) {
+    this.datastore = datastore;
+  }
+
+  public Long addPrivateComment(PrivateCommentData privateComment) {
+    Entity entity = privateComment.toEntity();
+    Key key = datastore.put(entity);
+    long commentID = key.getId();
+    privateComment.setCommentID(commentID);
+    return commentID;
+  }
+
+  public PrivateCommentData getPrivateCommentByID(long commentID) {
+    Query query =
+        new Query("PrivateComment")
+            .addFilter(
+                "__key__", FilterOperator.EQUAL, KeyFactory.createKey("PrivateComment", commentID));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    PrivateCommentData privateComment = new PrivateCommentData(entity);
+    return privateComment;
+  }
+
+  // only 1 private comment per task
+  public PrivateCommentData getPrivateCommentByTaskID(long taskID) {
+    Filter filter = new FilterPredicate("taskID", FilterOperator.EQUAL, taskID);
+    // private comments should be size = 1
+    ArrayList<PrivateCommentData> privateComments =
+        getPrivateComments(filter, NO_QUERY_LIMIT, NO_QUERY_SORT);
+    return privateComments.get(0);
+  }
+
+  public ArrayList<PrivateCommentData> getPrivateComments(
+      int limit, String sortBy, String sortDirection) {
+    SortPredicate sort =
+        new SortPredicate(sortBy, SortDirection.valueOf(sortDirection.toUpperCase()));
+    return getPrivateComments(NO_QUERY_FILTER, limit, sort);
+  }
```

Comment:
Is there a use case for when users would want to sort the private comment? Wouldn't a user only be possibly sorting their tasks and the private comment would just be displayed underneath/beside each task? Unless I'm misunderstanding how the private comments will be displayed?

Reply:
Yeah you're correct. Just wanted to include this in case the buttons next to the comments looked too clunky and I wanted to display only the comments (and make them buttons to tasks using href)